log/sequencer_test.go

```diff
@@ -235,6 +241,29 @@ func TestSequenceWithNothingQueued(t *testing.T) {
 	}
 }
 
+// Tests that the guard interval is being passed to storage correctly. Actual operation of the
+// window is tested by storage tests.
+func TestGuardWindowPassthrough(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	guardInterval := time.Second * 10
+	expectedCutoffTime := fakeTimeForTest.Add(-guardInterval)
+	params := testParameters{dequeueLimit: 1, shouldCommit: true, latestSignedRoot: &testRoot16, dequeuedLeaves: []trillian.LogLeaf{}, skipStoreSignedRoot: true, overrideDequeueTime:&expectedCutoffTime}
+
+	c := createTestContext(ctrl, params)
+	c.sequencer.SetGuardWindow(guardInterval)
+
+	leaves, err := c.sequencer.SequenceBatch(1, rootNeverExpiresFunc)
+	if leaves != 0 {
+		t.Fatalf("Expected no leaves sequenced when in guard interval but got: %d", leaves)
```

Comment:
AIUI the (strong) preference in Go tests is for a terser error message of a form like `"SequenceBatch(1)=%d,%q; want 0,nil", leaves, err`.  Given that this error is consistent with the rest of the file it's not worth changing now, but at some point we should align with normal Go practice.